<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head>
<link rel="stylesheet" href="stylesheets/styles.css">
<link rel="stylesheet" href="stylesheets/pygment_trac.css">
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

<meta http-equiv="Content-type" content="text/html; charset=UTF-8">
<meta http-equiv="Content-Style-Type" content="text/css">
<title>UFV2019</title>
</head>
<body>
<p class="menublock"><a href="http://morotalab.org/teaching/teaching.html">Back</a></p>

<h1>High-throughput Phenotyping Driven Quantitative Genetics @CMA-FCT-NOVA</h1>
<h2>October 20 and 22, 2021</h2>


<br>
<br>
<h3>Instructor</h3>
<ul>
  <li>Gota Morota - Virginia Polytechnic Institute and State University</li>
</ul>

<br>
<br>
<h3>Date and Location</h3>
<ul>
  <li>October 20 and 22, 2021</li>
  <li>NOVA School of Science and Technology at the NOVA University of Lisbon, Caparica, Portugal</li>
</ul>




<br>
<br>
<h3>Course description</h3>
This course will cover quantitative genetic analysis of complex trait genomics with an emphasis on image-derived high-throughput phenotyping data.
The advent of plant phenomics, coupled with the wealth of genotypic
data generated by next-generation sequencing technologies, provides
exciting new resources for investigations into and improvement of
complex traits. However, these new technologies also bring new
challenges in quantitative genetics, namely, a need for the
development of robust frameworks that can accommodate these
high-dimensional data. We will discuss statistical methodologies for connecting phenotypes with high-dimensional genomic information to better understand polygenic traits from prediction and inference
perspectives. Topics will include phenotyping, genomic relatedness,
linkage disequilibrium, population stratification, genomic
heritability, genome-wide association study, genomic prediction,
causal inference, and statistical learning. We will use examples from
the plant and animal genetics literature.

This is a two-day course. Each day will include lectures, hands-on
data analysis sessions, and class discussions. Hands-on sessions will
involve the data analysis of simulated and real genomic data available
in public repositories. The course will use R/RStudio and Julia for
statistical computing tools. The course is aimed at students,
researchers, and professionals interested in the quantitative genetic
analysis of high-throughput phenotyping data. Some basic understanding
of genetics, statistics, and programming will be beneficial.



<br>
<br>
<h3>Schedule</h3>
<ul>

  <li>October 20 (Wed.)</li>
  <ul style="list-style-type:none">
    <li>10:00  -11:00: Introduction to Genome to Phenome - lecture [<a href="http://morotalab.org/UFV2019/day1/day1.html">HTML</a>][<a href="http://morotalab.org/UFV2019/lab1/lab1.html">R</a>]</li>
  <li>11:00 -12:00: Decoding best linear unbiased prediction (BLUP) - lecture </li>
  <li>12:00 -13:30 pm: Lunch break</li>
  <li>13:30 -14:30 pm: Decoding best linear unbiased prediction (BLUP) - hands-on </li>
  <li>14:30 -15:30 pm: Decoding best linear unbiased prediction (BLUP) - hands-on</li>
  </ul>


  <li>October 22 (Fri.)</li>
   <ul style="list-style-type:none">
    <li>10:00  -11:00: Deterministic formulas for genomic prediction - lecture and hands-on </li>
  <li>11:00 -12:00: Statistical methods for the quantitative genetic analysis of
high-throughput phenotyping data - lecture </li>
  <li>12:00 -13:30: Lunch break</li>
  <li>13:30 -14:30: Statistical methods for the quantitative genetic analysis of
high-throughput phenotyping data - lecture </li>
  <li>14:30 -15:30: Statistical methods for the quantitative genetic analysis of
high-throughput phenotyping data - hands-on</li>
  </ul>
  
</ul>

  

  



</body>
</html>
